### DISCUSSION TITLE
HomeAssistant / RaspberryMatic / Integration: "keine Verbindung"

### DISCUSSION THREAD

**Discussion by bsafh** [OpenCCU/OpenCCU#1850] [🇩🇪 Fragen&Antworten]
Hallo,

ich habe HomeAssistant supervised laufen in einer VM unter kvm (auf einer TrueNAS Scale). In HomeAssistant habe ich das RaspberryMatic Addon eingebunden.

Klasse Sache, wenn's läuft, denke ich mir - es geht um 19 Homematic (classic) Heizkörperthermostate.

Nun scheitere ich aber am letzten Schritt von https://github.com/jens-maus/RaspberryMatic/wiki/HomeAssistant-Integration 

Es kommt nur "keine Verbindung".
Im RaspberryMatic Docker container ist die Fireall auf relaxed, alle Ports sind offen - was ein nmap-scan vom Basis Debian auch zeigt

root@homeassistant:~# nmap 172.30.33.1
Starting Nmap 7.80 ( https://nmap.org ) at 2022-05-27 19:34 CEST
Nmap scan report for 172.30.33.1
Host is up (0.000030s latency).
Not shown: 991 closed ports
PORT     STATE    SERVICE
22/tcp   filtered ssh
80/tcp   open     http
443/tcp  open     https
1999/tcp open     tcp-id-port
2000/tcp open     cisco-sccp
2001/tcp open     dc
2010/tcp open     search
8099/tcp open     unknown
8181/tcp open     intermapper
MAC Address: 02:42:AC:1E:21:01 (Unknown)

![2022-05-27 19_41_51-RaspberryMatic – Home Assistant - 192 168 80 21_ - http___192 168 80 21_8123_](https://user-images.githubusercontent.com/106388187/170763296-47d3386b-85a0-4973-9dbe-888ac9414839.png)

Der Name des Docker Containers stimmt auch (mit der IP geht's aber auch nicht)
![2022-05-27 19_43_22-Home Assistant - 192 168 80 21_ - http___192 168 80 21_8123_](https://user-images.githubusercontent.com/106388187/170763450-2faead89-b621-4542-8298-785530725211.png)

![2022-05-27 19_44_26-Einstellungen – Home Assistant - 192 168 80 21_ - http___192 168 80 21_8123_](https://user-images.githubusercontent.com/106388187/170763588-aaa086b4-1f81-4573-bc34-3bbffba58303.png)

In der Konfiguration des RaspberryMatic Addons/Docker Containers sind auch alle Ports (nicht nur die hier gezeigten, aber das sind ja die, die in der Konfiguration der Integration auftauchen) aktiviert:
![2022-05-27 19_45_27-Home Assistant - 192 168 80 21_ - http___192 168 80 21_8123_](https://user-images.githubusercontent.com/106388187/170763885-80236882-493e-4054-ab21-b44078059bec.png)

Passwort stimmt auch (per copy&paste übertragen)

**Jetzt gehen mir wirklich die Ideen aus - was ist das noch falsch?**

(Anbindung an die reale Welt soll mit HB-RF-ETH+dem neuen Funkmodul (ich kann mir die Bezeichnung nie merken) gehen - das zickts aber noch - anderes Thema (HB-RF-ETH nach Firmware-Update gebrickt...)

Versionen:
RaspberryMatic CCU
Current version: 3.63.9.20220521

custom_homematic 1.5.4

Jede Hilfe ist willkommen .. :-(

Home Assistant Core 2022.5.5
Home Assistant Supervisor 2022.05.3
Kernel version 5.10.0-13-amd64
Agent version 1.2.2

**Comment by Baxxy13** [comment]
Wenn deine RaspberryMatic kein Funkmodul hat läuft auch keiner der Dienste. (HmIP-RF, BidCos-RF)
Läuft kein Dienst kann sich die Integration nicht verbinden.

**Reply by bsafh** [reply]
OK...

Dann muss ich erst einmal das Problem mit dem Funkmodul lösen.

Das HB-RF-ETH ist per ping erreichbar, die Weboberfläche ist auch erreichbar, aber der avahi-browser findet es nicht - genauer: er findet gar nichts außer den lokal auf dem Rechner installierten Diensten.

Auch wenn ich die IP-Adresse in /etc/default/hb-rf-eth selbst eintrage (was sonst das postinstall Skript des Debian hb-rf-eth Pakets tut), kommt keine Verbindung zustande, die RaspberryMatic (also das Addon in HA) findet es nicht ... :-(

Fragt sich nun, ob das an der lokalen Konfiguration auf dem HA-supervised ist - oder ob es irgendwo im Netzwerk klemmt (da sind Switche dazwischen, aber es ist alles in einem Subnet). (das ist ein blödes Puzzle ...)

**Reply by Baxxy13** [reply]
Die "Kontroll-Datei" muss innerhalb RaspberryMatic in /etc/config/ liegen.
Kann aber nicht sagen ob die Funktionalität mit dem RM-HA-AddOn allgemein gegeben ist. 
Vielleicht das Setup erstmal mit ner RM standalone auf einem Pi testen.

**Reply by bsafh** [reply]
erstmal haben, den Pi :-)

Ich habe praktisch alles virtualisiert.

Aber es gibt eine Teil-Auflösung:

1) im Backbone Switch (Avaya 4548GT, falls es wen interessiert) war igmp snooping an, aber dann blockt er alle anderen Multicast-Adressen - auch die für mDNS nötige 224.0.0.251. Nachdem ich die mit einem obskuren CLI Befehl (`vlan igmp allow-<irgeendwas>-flood <vlan> <multicast-ip>` oder so) freigegeben hatte, erschien der RF-HB-ETH schon mal in der Liste von `mdns-scan` - aber nur auf dem Host der VMs (kvm/qemu auf TrueNAS Scale (Debian 11) - nicht in der virtuellen Maschine selbst :-(

Erst ein `/usr/sbin/ip link set dev macvtap0 allmulticast on` hat dann bewirkt, dass `mdns-scan` und `avahi-browse -a` auch in der VM (also dem HA-supervised) das Funkmodul anzeigen.

Dann geht auch die Installation des Debian-Pakets dafür - auf dem HA-supervised-Debian in der VM.

Ob das nun auch noch im RM-HA-Addon ankommt - keine Ahnung. Das ist ja ein Docker-Container, wie ich das da rein leite - mal sehen. "Automatisch" geht das bestimmt nicht.

Wie muss denn die "Kontroll-Datei" heißen und aussehen? Im HA-supervised sieht das so aus:
```
root@homeassistant:/etc# cat /etc/default/hb_rf_eth
HB_RF_ETH_ADDRESS="192.168.80.129"
```

Wenn das auch im Docker-Container so sein muss, kann man das ja dahin einblenden - irgendwie (also, zumindest mit docker-compose, aber das Docker-YAML dafür hat man ja nicht).

Dumm ist nur, dass der HB-RF-ETH nach ein paar Minuten einfach stirbt (manchmal) und dann auch nicht per ping mehr erreicht werden kann - da hilft dann nur Neustart. Dann kann ihn natürlich auch die RM nicht mehr finden :-(

**Reply by bsafh** [reply]
Next step  ....

HB-RF-ETH über Nacht aus, heute morgen wieder angemacht, gleiches Problem. 30s Kontakt (ping), dann tot

weiterlaufen lassen...
ca 5 Std später mal zufällig geguckt: nanu, ist ja jetzt erreichbar per ping
und zwar auch aus dem docker-container des RM-HA Addons  (`docker exec -it -u root <containername> /bin/sh` und dann halt `ping`)

also flugs die IP in `/etc/config/hb_rf_eth` eingetragen und docker container neu gestartet (aus HomeAssistant heraus) und die Weboberfläche des RM-HA-Addons in HA aufgerufen

Ergebnis:
`502 Bad Gateway`

 `/etc/config/hb_rf_eth` umbenannt, nochmal RB-HA-Addon neu gestartet: Oberfläche ist wieder da. Nanu?

 `/etc/config/hb_rf_eth` wieder zurück kopiert, neu gestartet
`502 Bad Gateway`

Gut, dann mal ins Logfile schauen (im Container)

```
cat /var/log/boot.log

IP Adresse isn RaspberryMatic-Addon eingetragen, docker neu gestartet und...
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.1, OK
Identifying Homematic RF-Hardware: ...../etc/init.d/S47InitRFHardware: line 303: can't create /sys/bus/platform/drivers/leds-gpio/unbind: nonexistent directory
/etc/init.d/S47InitRFHardware: line 314: can't create /sys/bus/platform/drivers/leds-gpio/bind: nonexistent directory
HmRF: RPI-RF-MOD/HB-RF-ETH@192.168.80.129, HmIP: RPI-RF-MOD/HB-RF-ETH@192.168.80.129, OK
Updating Homematic RF-Hardware: RPI-RF-MOD: 3.4.2=>4.4.22, ERROR (0.0.0 != 4.4.22)
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: ....................ERROR
Starting HMIPServer: .......................................................................................................................................................ERROR
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.63.9.20220521 (raspmatic_oci_amd64)
```
In HomeAssistant allerdings kommt trotzdem nur "502 Bad Gateway"

**Hmm, HB-RF-ETH gefunden! :-)**

**HmRF: RPI-RF-MOD/HB-RF-ETH@192.168.80.129, HmIP: RPI-RF-MOD/HB-RF-ETH@192.168.80.129, OK**
**Updating Homematic RF-Hardware: RPI-RF-MOD: 3.4.2=>4.4.22, ERROR (0.0.0 != 4.4.22)**

**Aber woher kommt jetzt das "`502 Bad Gateway`"? **

**Und zwar nur mit dem eingebundenen HB-RF-ETH, ohne kommt es nicht?**

**Reply by Baxxy13** [reply]
Puuh, Sorry. So tief stecke ich in der Materie nicht drin um dir weiterzuhelfen.
Wenn man Querbeet liest findet man immer wieder Hinweise das das RM-HA-AddOn in einer Supervised Installation nicht korrekt funktioniert. Empfohlen wird Homeassistant-OS zu nutzen oder RM in eine VM zu verfrachten.

**Comment by bsafh** [comment]
Hmpf.

Dann funktioniert aber die Integration in HomeAssistant nicht mehr wirklich - dann kann ich zB keine Zigbee-Fensterkontakte nutzen, um die Homematic-Ventile schließen zu lassen. 

die Datei muss übrigens nur die nackte IP enthalten, das habe ich jetzt heraus gefunden
Name: /etc/config/hb-rf-eth
Inhalt: 192.168.x.x

Lief auch einmal - die RM erkannte das Funkmodul, lies sich in den Anlernmodus versetzen, Ventil ebenfalls in Anlernen.

Ergebnis:
Ventil sagt "AC", also erfolgreich angelernt.

RM sagt: nix im Posteingang, nichts angelernt.

Und so blieb es dann.

Zwischendurch einmal das Funkmodul an einen anderen Ort in der Wohnung gebracht, dabei natürlich stromlos gemacht.

Ergebnis: 
jetzt erkennt das HB-RF-ETH die auf dem Modul selbst aufgesteckte Funkplatine nicht mehr (obwohl ja die Stromversorgung erfolgreich über eben jenes Funkmodul läuft). (taucht in der Weboberfläche des ETH-Moduls nicht mehr auf, da steht nur "-" als Funkmodul). Auch nicht nach 10x stromlos machen.

Dann kann die RM natürlich das Funkmodul auch nicht sehen. :-(

Riecht irgendwie gerade nach Hardware-Fehler (zusätzlich zu allem anderen)

**Reply by Baxxy13** [reply]
> Dann funktioniert aber die Integration in HomeAssistant nicht mehr wirklich - dann kann ich zB keine Zigbee-Fensterkontakte nutzen, um die Homematic-Ventile schließen zu lassen.

Verstehe den Zusammenhang nicht. Homeassistant ist es doch völlig egal wo RM residiert. Die Integration sorgt doch dafür das alles in HA verfügbar ist.



> die Datei muss übrigens nur die nackte IP enthalten, das habe ich jetzt heraus gefunden

Sowas muss man doch nicht herausfinden, steht doch im [Wiki](https://github.com/jens-maus/RaspberryMatic/wiki/Experten-Features#hb-rf-eth-anbindung) beschrieben.


> Riecht irgendwie gerade nach Hardware-Fehler (zusätzlich zu allem anderen)

Könnte auch ein fehlgeschagenes Firmware-Update des RPI-RF-MOD sein. Fragen dazu am besten im Homematic-Forum [Platine HB-RF-ETH: Funkmodul per Netzwerk anbinden](https://homematic-forum.de/forum/viewtopic.php?f=76&t=60150).

**Reply by bsafh** [reply]
Wiki finden ist auch "herausbekommen" ...

Die Integration in HA wird als AddOn (also als Docker-Container) für HA bereitgestellt. 

Ob die eigentliche "Integration" (in HA-Sprache) auch mit einer abgesetzten RM (zB in einer anderen VM) läuft - wäre natürlich auch eine Möglichkeit, wenne s geht (geht halt nur etwas Komfort verloren), ich hab's aber nicht gefunden

Fehlgeschlagenes Firmware-Update schließe ich erst einmal aus, weil die Kommunikation zur RM mit dem nur halb geglückten Anlernen überhaupt erst nach einem Firmware-Update ging, vorher kam gar keine Verbindung zustande.

Und nach dem Update wurde auch (bis zum letzten Reboot) in der WegGUI des ETH Moduls das Funkmodul angezeigt. (vorher nicht)

**Reply by Baxxy13** [reply]
Die "Integration" braucht nur Netzwerkzugriff auf die Laufende CCU/RM/piVCCU/debMATIC. Eingerichtet wird idealerweise über die IP-Adresse. 
Wo die laufen ist völlig Schnuppe. 
In meinem Test-Zoo sind aktuell 3 RM's in HA integriert. Eine VM-RM, eine TinkerS-RM und eine Pi3B+-RM.

**Comment by blade122333** [comment]
Es liegt am Bindestrich im Hostnamen, es muß  ein unterstrich sein.
